shimmingtoolbox/pmu.py

```diff
@@ -123,29 +123,66 @@ def read_resp(self, fname_pmu):
 
         return attributes
 
-    def get_times(self):
+    def get_times(self, start_time=None, stop_time=None):
         """
         Get the times in ms at which the respiration took place.
 
+            start_time (int): Start time in milliseconds past midnight
+            stop_time (int): Stop time in milliseconds past midnight
+
         Returns:
             np.ndarray: Array containing the timepoints in ms of each data
         """
-        raster = float(self.stop_time_mdh - self.start_time_mdh) / (len(self.data)-1)
-        times = (self.start_time_mdh + raster * np.arange(len(self.data)))  # ms
+        times = self._get_all_times()
+        start_idx, stop_idx = self._get_time_indexes(start_time, stop_time)
+
+        # +1 is needed to include the stop_idx
+        return times[start_idx:stop_idx + 1]
 
+    def _get_all_times(self):
+        raster = float(self.stop_time_mdh - self.start_time_mdh) / (len(self.data) - 1)
```

Comment:
make sure MDH refers to the timing of the ACQUISITION (not the recon)-- add documentation in the code